pkg/gather/service/analyze.go

```diff
@@ -47,36 +53,51 @@ func analyzeGatherBundle(bundleFile io.Reader) error {
 		if header.Typeflag != tar.TypeReg {
 			continue
 		}
-		filenameParts := strings.SplitN(header.Name, "/", 2)
-		if len(filenameParts) != 2 {
+
+		serviceEntriesFileSubmatch := serviceEntriesFilePathRegex.FindStringSubmatch(header.Name)
+		if serviceEntriesFileSubmatch == nil {
 			continue
 		}
-		// we only care about the release-image.service for now. in the future, we will look at other services, too.
-		if filenameParts[1] == "bootstrap/services/release-image.json" {
-			var err error
-			releaseImageAnalysis, err = analyzeService(tarReader)
-			if err != nil {
-				logrus.Infof("Could not analyze the release-image.service: %v", err)
-			}
-			break
+		serviceName := serviceEntriesFileSubmatch[1]
+
+		serviceAnalysis, err := analyzeService(tarReader)
+		if err != nil {
+			logrus.Infof("Could not analyze the %s.service: %v", serviceName, err)
+			continue
 		}
+
+		serviceAnalyses[serviceName] = serviceAnalysis
 	}
 
-	// log details about the release-image.service.
-	if releaseImageAnalysis != nil && releaseImageAnalysis.starts > 0 {
-		if !releaseImageAnalysis.successful {
-			logrus.Error("The bootstrap machine failed to download the release image")
-			for _, l := range strings.Split(releaseImageAnalysis.lastError, "\n") {
-				logrus.Info(l)
-			}
+	analysisChecks := []struct {
+		name  string
+		check func(analysis) bool
+	}{
+		{name: "release-image", check: checkReleaseImageDownload},
+	}
+	for _, check := range analysisChecks {
+		a := serviceAnalyses[check.name]
+		if a.starts == 0 {
+			logrus.Errorf("The bootstrap machine did not execute the %s.service systemd unit", check.name)
+			break
+		}
+		if !check.check(a) {
+			break
```

Comment:
This `break` too. IIUC these seem like optimizations based on the limited number of checks, which is fine for the time being.

Reply:
Same response as above, where this break is what we want long-term.